src/client.rs

```diff
@@ -0,0 +1,149 @@
+use std::{io::stdin, str::FromStr, thread, time::Duration};
+
+use reqwest::Client;
+use tictactoe::{Game, Mark, State};
+use uuid::Uuid;
+
+const SERVER_URL: &str = "http://127.0.0.1:80/game";
+
+#[tokio::main]
+async fn main() {
+    let client = Client::new();
+    let game_id = get_or_create_game(&client).await;
+    let player_role = get_player_role();
+
+    loop {
+        let game_url = format!("{SERVER_URL}/{}", game_id.to_string());
+        let game = get_game(&client, game_url.as_str()).await;
+
+        match game.state() {
+            State::Playing(active_role) => {
+                if active_role != player_role {
+                    thread::sleep(Duration::from_secs(1));
+                    continue;
+                }
+
+                println!("{}", game.board());
+
+                let (row, column) = get_move();
+                post_move(&client, game_url.as_str(), row, column).await;
+                let game = get_game(&client, game_url.as_str()).await;
+
+                println!("{}", game.board());
+            }
+            State::Win(winner) => {
+                println!("{winner} wins!");
+                return;
+            }
+            State::Tie => {
+                println!("Tie.");
+                return;
+            }
+        }
+    }
+}
+
+async fn get_or_create_game(client: &Client) -> Uuid {
+    println!("Create game (1) or join existing one (2)?");
+    let mut buffer = String::default();
+    stdin()
+        .read_line(&mut buffer)
+        .expect("failed to read the choice");
+    let choice: u32 = buffer
+        .trim()
+        .parse()
+        .expect(format!("expected a number, got {buffer}").as_str());
+
+    let game_id: Uuid = match choice {
+        1 => {
```

Comment:
magic constant antipattern; create variables NEW_GAME and JOIN_EXISITING_GAME for better understanding of what is happenning